esignet-core/src/main/java/io/mosip/esignet/core/validator/PKCECodeChallengeMethodValidator.java

```diff
@@ -0,0 +1,21 @@
+package io.mosip.esignet.core.validator;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.List;
+
+@Component
+public class PKCECodeChallengeMethodValidator implements ConstraintValidator<PKCECodeChallengeMethod, String> {
+
+    @Value("#{${mosip.esignet.supported-pkce-methods}}")
+    private List<String> supportedMethods;
+
+    @Override
+    public boolean isValid(String s, ConstraintValidatorContext constraintValidatorContext) {
+        return (!StringUtils.isEmpty(s) && !supportedMethods.contains("S256")) ? false : true;
```

Comment:
why is S256 hardcoded here, should this not be the s variable, also please don't have strings with single letters

Reply:
Fixed